### PR TITLE
Use new serialization for events and check

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?

Serialize payloads by building them incrementally for events and service checks.

### Motivation

The old split serialization logic, currently still used by those 2 payloads, was not checking the uncompressed size.

### Additional Notes

Reviewing commit by commit can help

### CODE REVIEW

File: pkg/releasenotes/notes/use-new-serialization-for-events-and-servicechecks-2b9d2f1331444637.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    Fix 'Error code "413 Request Entity Too Large" received while sending transaction to "https://dd.agent.datadoghq.eu/api/v1/check_run?api_key=*************************28e95": Payload too big, dropping it'
```
Comment (hush-hush) [inline pkg/releasenotes/notes/use-new-serialization-for-events-and-servicechecks-2b9d2f1331444637.yaml]: This is not a very clear message for user to see in the changelog. We should better decribe the issue and th fix I think.

What about somethinf like:

```
Add options to enable stream serialization for `events` and `service_check` in order to fix error 'Request Entity Too Large' when sending payload to Datadog.
```
?

File: pkg/metrics/event.go [hunk 2]
```diff
@@ -175,3 +183,144 @@ func (events Events) SplitPayload(times int) ([]marshaler.Marshaler, error) {
...
+		stream.WriteArrayStart()
+	}
+
+	if err := writeEvent(event, stream); err != nil {
+		return err
+	}
+	return stream.Flush()
+}
+
+// SupportJSONSeparatorInsertion returns true to add JSON separator automatically between two calls of WriteItem, false otherwise.
```
Comment (hush-hush) [inline pkg/metrics/event.go]: I don't find the name `SupportJSONSeparatorInsertion` very explicit for this function. `AddJSONSeparatoraAutomatically` ?

**Review by hush-hush — COMMENTED** [review]
Looks good overall but I might have miss some details. I think we should wait for a review from @arbll .

File: pkg/releasenotes/notes/use-new-serialization-for-events-and-servicechecks-2b9d2f1331444637.yaml (hunk 1 above)
Comment (ogaca-dd) [reply pkg/releasenotes/notes/use-new-serialization-for-events-and-servicechecks-2b9d2f1331444637.yaml]: The Trello card was created after a customer issue who had 'Error code "413 Request Entity Too Large'. Should I add two lines, one as fixe and another one as improvement or one is enough?
What do you think?

**Comment by ogaca-dd** [comment]
Closed as the logic will be simplified: 
- ServiceCheck: https://github.com/DataDog/datadog-agent/pull/4016
- Events: Coming soon.

**Pull request closed without merging**